syntax = "proto3";

message ClientMessage {
  string operation = 1; // "read" or "write"
  string query = 2;     // exactly 8 characters expected
  bytes data = 3;
}

message RoutedMessage {
  string client_id = 1;
  ClientMessage client_message = 2;
}

message FileChunk {
  string client_id = 1;
  uint32    index     = 2;
  bool      last      = 3;
  bytes     payload   = 4;
}